src/snowcli/cli/nativeapp/manager.py

```diff
@@ -228,6 +229,15 @@ def app_role(self) -> str:
     def debug_mode(self) -> bool:
         return self.definition.get("application", {}).get("debug", True)
 
+    @cached_property
```

Comment:
what does caching do in this case? does any file change invalidate the cache?

Reply:
Yes, any file change would invalidate the cache. But the lifetime of a SnowCLI execution is only the command, so we're not interested in picking up changes that might happen during the few seconds we're executing.